dissect/target/plugins/apps/browser/firefox.py

```diff
@@ -305,6 +311,76 @@ def downloads(self) -> Iterator[BrowserDownloadRecord]:
+                ts_update (datetime): Extension update timestamp.
+                browser (string): The browser from which the records are generated.
+                id (string): Extension unique identifier.
+                name (string): Name of the extension.
+                short_name (string): Short name of the extension.
+                default_title (string): Default title of the extension.
+                description (string): Description of the extension.
+                version (string): Version of the extension.
+                ext_path (path): Relative path of the extension.
+                from_webstore (boolean): Extension from webstore.
+                permissions (string[]): Permissions of the extension.
+                manifest (varint): Version of the extensions' manifest.
+                optionalPermissions (string[]): Optional permissions of the extension.
+                sourceURI (path): Source path from which the extension was downloaded.
+                source (path): The source file of the download record.
+        """
+        for user, _, profile_dir in self._iter_profiles():
+            extension_file = profile_dir.joinpath("extensions.json")
+
+            if not extension_file.exists():
+                self.target.log.warning(
+                    "No 'extensions.json' addon file found for user %s in directory %s", user, profile_dir
+                )
+                continue
+
+            try:
+                extensions = json.load(extension_file.open())
+
+                for extension in extensions.get("addons", []):
+                    try:
+                        yield self.BrowserExtensionRecord(
+                            ts_install=extension.get("installDate", 0) // 1000,
+                            ts_update=extension.get("updateDate", 0) // 1000,
+                            browser="firefox",
+                            id=extension.get("id"),
+                            name=extension.get("defaultLocale").get("name"),
+                            short_name=None,
+                            default_title=None,
+                            description=extension.get("defaultLocale").get("description"),
+                            version=extension.get("version"),
+                            ext_path=extension.get("path"),
+                            from_webstore=None,
+                            permissions=extension.get("userPermissions").get("permissions"),
+                            manifest_version=extension.get("manifestVersion"),
+                            sourceURI=extension.get("sourceURI"),
+                            optionalPermissions=extension.get("optionalPermissions").get("permissions"),
+                            source=extension_file,
+                            _target=self.target,
+                            _user=user.user,
+                        )
+                    except (AttributeError, KeyError) as e:
```

Comment:
```suggestion
                    except (AttributeError, KeyError) as e:
```
Maybe good to double check but with the proposed changes above, I don't think these can happen anymore.